src/features/classification/Classification.js

```diff
@@ -1,25 +1,48 @@
 // @flow
 import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
 
-import Tooltip from '../../components/tooltip';
+import ClassifiedBadge from './ClassifiedBadge';
+import AddClassificationBadge from './AddClassificationBadge';
+import messages from './messages';
 import './Classification.scss';
 
+const STYLE_INLINE: 'inline' = 'inline';
+const STYLE_TOOLTIP: 'tooltip' = 'tooltip';
+
 type Props = {
     advisoryMessage?: string,
-    isMessageInline?: boolean,
-    name: string,
+    className?: string,
+    messageStyle?: typeof STYLE_INLINE | typeof STYLE_TOOLTIP,
+    name?: string,
 };
 
-const Classification = ({ advisoryMessage, isMessageInline = true, name }: Props) => {
+const Classification = ({ advisoryMessage, className = '', messageStyle, name }: Props) => {
+    const isClassified = !!name;
     const hasMessage = !!advisoryMessage;
-    const isTooltipEnabled = hasMessage && !isMessageInline;
-    const isInlineMessageVisible = hasMessage && isMessageInline;
+
+    const isTooltipMessageEnabled = isClassified && hasMessage && messageStyle === STYLE_TOOLTIP;
+    const isInlineMessageEnabled = isClassified && hasMessage && messageStyle === STYLE_INLINE;
+
+    // Either the add classification badge should be visible or the "not classified" text or neither
+    const isNotClassifiedBadgeVisible = !isClassified && !messageStyle;
```

Comment:
Just curious as to why this depends on `messageStyle`. Name might also be a bit clearer as `isClassifiedBadgeHidden`.

Reply:
Gonna change it to isAddClassificationBadgeVisible